### PR TITLE
consul: 0.9.3 -> 1.3.0 with vendored UI

### PR DESCRIPTION
**Pull request by nh2**
###### Motivation for this change

Subsumes #48714, #44192, #41243, and #35602.

The key point is that we no longer have to build the Consul UI, it is [vendored](https://github.com/NixOS/nixpkgs/pull/48714#issuecomment-433454834) into the binary.

See commit messages for details.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by nh2** [comment]
CC @arianvp @vdemeester @c0bw3b

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

File: pkgs/servers/consul/default.nix [hunk 1]
```diff
@@ -1,22 +1,26 @@
-{ stdenv, buildGoPackage, consul-ui, fetchFromGitHub }:
+{ stdenv, buildGoPackage, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGo110Package rec {
```
Comment (arianvp) [inline pkgs/servers/consul/default.nix]: IS this a typo?

Comment (arianvp) [reply pkgs/servers/consul/default.nix]: if not, I don't understand where this identifier comes from :)

**Comment by nh2** [comment]
Btw, my usage of

* `python-consul`
* `consul-alerts`

in my staging deployment suggests that the they work well with this Consul upgrade.

File: pkgs/servers/consul/default.nix (hunk 1 above)
Comment (nh2) [reply pkgs/servers/consul/default.nix]: Typo from my forward-port, fixed

**Comment by c0bw3b** [comment]
And with real chunks of documentation on top of it :tada:

File: pkgs/servers/consul/default.nix [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
   name = "consul-${version}";
-  version = "0.9.3";
+  version = "1.3.0";
   rev = "v${version}";
 
   goPackagePath = "github.com/hashicorp/consul";
 
+  # Note: Currently only release tags are supported, because they have the Consul UI
+  # vendored. See
+  #   https://github.com/NixOS/nixpkgs/pull/48714#issuecomment-433454834
```
Comment (arianvp) [inline pkgs/servers/consul/default.nix]: Perhaps add a link to https://github.com/NixOS/nixpkgs/pull/49082  here, so people know how such a build would look like?

**Review by arianvp — CHANGES_REQUESTED** [review]

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build consul

**Review by c0bw3b — APPROVED** [review]

**Review by arianvp — APPROVED** [review]

**Comment by c0bw3b** [comment]
@nh2 could you rebase and solve the conflict with the release note XML please?

**Comment by arianvp** [comment]
I have one small change requests that we can add:

- [ ] Set `Type=notify` on the systemd unit. Apparently this is supported. 
  > When running under systemd on Linux, Consul notifies systemd by sending READY=1 to the $NOTIFY_SOCKET when a LAN join has completed. For this either the join or retry_join option has to be set and the service definition file has to have Type=notify set.
  https://www.consul.io/docs/agent/basics.html#running-an-agent

**Comment by nh2** [comment]
@arianvp Let's do that as part of a separate PR or issue, as this will require more testing.

**Comment by nh2** [comment]
Rebased to fix release notes merge conflict

**Comment by nh2** [comment]
> I have one small change requests that we can add:

> @arianvp Let's do that as part of a separate PR or issue, as this will require more testing.

Extracted as https://github.com/NixOS/nixpkgs/issues/49706

**Pull request merged**